conu/utils/probes.py

```diff
@@ -0,0 +1,120 @@
+import time
+import logging
+from multiprocessing import Process, Queue
+
+logger = logging.getLogger(__name__)
+
+
+class Probe(object):
+    """
+    Probe can be used for waiting on specific result of a function.
+    Probe ends when function returns expected_retval or timeout is exceeded.
+
+    Attributes:
+        timeout              Number of seconds spent on trying. Set timeout to -1 for infinite run.
+        pause                Number of seconds waited between multiple function result checks
+        expected_exceptions  When one of expected_exception is raised, probe ignores it and tries to run function again.
+                             To ignore multiple exceptions use parenthesized tuple.
+        expected_retval      When expected_retval is recieved, probe ends successfully
+        fnc                  Function which run is checked by probe
+    """
+    def __init__(self,
+                 timeout=1,
+                 pause=1,
+                 expected_exceptions=(),
+                 expected_retval=True,
+                 fnc=bool,
+                 **kwargs):
+
+        self.timeout = timeout
+        self.pause = pause
+        self.expected_exceptions = expected_exceptions
+        self.fnc = fnc
+        self.kwargs = kwargs
+        self.expected_retval = expected_retval
+        self.process = None
+        self.queue = None
+
+    def run(self):
+        if self.process and self.process.is_alive():
+            raise RuntimeError("One instance of probe can only be started once")
```

Comment:
Should the message rather say `One instance of Probe can only be probing once at any given time` or something like that?